tests/contrib/io/matplotlib/test_simple_io.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited (“QuantumBlack”) name and logo
+# (either separately or in combination, “QuantumBlack Trademarks”) are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import matplotlib
+import numpy as np
+import pytest
+
+from kedro.contrib.io.matplotlib import MatplotlibWriter
+from kedro.io import DataSetError
+
+matplotlib.use('TkAgg')  # used to facilitate simple inclusion into kedro CI/CD
+
+
+def test_should_write_to_image_in_single_mode(tmpdir):
+    import matplotlib.pyplot as plt
+
+    # generate plot
+    plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+
+    # write and compare
+    trusted_filepath = str(tmpdir.join("image_we_expect.png"))
+    plt.savefig(trusted_filepath)
+
+    experimental_filepath = str(tmpdir.join("image_we_write.png"))
+    plot_writer = MatplotlibWriter(filepath=experimental_filepath)
+    plot_writer.save(plt)
+
+    plt.close()
+
+    assert (
+        open(experimental_filepath, "rb").read() == open(trusted_filepath, "rb").read()
```

Comment:
What is the benefit of explicitly closing the open?

Reply:
It's considered good practice. If you don't explicitly close (or use a mechanism that does like `with open` or `Path.read_text_text`) then the file gets closed when the garbage collector finalizes it. While that is usually relatively soon, it's only actually guaranteed to happen before the process exits (and then only if the Python VM comes down cleanly). If you are writing to the file that's a quick path to data corruption. If you are reading it's not usually an issue unless you are on a filesystem that read locks, which is anything on Windows and some network things on other systems.

Reply:
fair enough, will implement

Reply:
Further if you don't close it because you know it's safe not to, reviewers have to think about whether it's safe, where as if you close it anyway they don't. Good practice is often about making the reviewers think less.

Reply:
Sounds good :)